File: 0485. Max Consecutive Ones/0485.java
// time complexity: O(n)
// space complexity: O(1)
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        int current_max = 0;
        int max = 0;

        for(int num : nums){
            if(num==1){
                current_max++; // increase the count if 1 is found
                if(current_max > max) max = current_max;
            }
            else current_max = 0; // reset the count if 0 is found
        }

        return max;
    }
}

File: 0485. Max Consecutive Ones/0485.py
# time complexity: O(n)
# space complexity: O(1)
class Solution(object):
    def findMaxConsecutiveOnes(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        current_max = 0
        maximum = 0

        for num in nums:
            if num == 1:
                current_max += 1
                maximum = max(maximum, current_max)
            else:
                current_max = 0

        return maximum